### PR TITLE
transition some docs to doctests

### LINKED ISSUE #1403: Non-reproducible example in documentation
In the documentation [here](https://github.com/FluxML/Flux.jl/blob/8fb94be4a682fe6879368d55c26e50359c049703/src/layers/basic.jl#L147), it appears that the dimensions don't match.

```
julia> import Flux

julia> m = Flux.Dense(10, 5)
Dense(10, 5)

julia> Flux.outdims(m, (5, 2)) == (5,)
ERROR: DimensionMismatch("input size should equal to (10,), got (5, 2)")
Stacktrace:
 [1] outdims(::Flux.Dense{typeof(identity),Array{Float32,2},Array{Float32,1}}, ::Tuple{Int64,Int64}) at /home/sid/.julia/packages/Flux/05b38/src/layers/basic.jl:152
 [2] top-level scope at REPL[10]:1

julia> 
```

### PR DESCRIPTION
**Pull request by gxyd**
Fixes #1403

### CODE REVIEW

File: src/layers/basic.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ on a given input.
 
 # Examples
 ```jldoctest
-julia> m = Chain(x -> x^2, x -> x+1);
+julia> m = Chain(x -> x^2, x -> x);
```
Comment (gxyd) [inline src/layers/basic.jl:12]: Will undo this, I made this change intentionally to see whether CI fails or not.

**Comment by gxyd** [comment]
Partially goes in the direction of #561 . This is ready now.

**Comment by DhairyaLGandhi** [comment]
Lgtm, thanks a lot!

bors r+

**Comment by gxyd** [comment]
Thanks @DhairyaLGandhi . Btw, you'll still need to approve 😉 .

**Review by CarloLucibello — APPROVED** [review]

**Pull request merged**